### PR TITLE
Ensure locked sample have an icon and are not editable.

### LINKED ISSUE #198: Locked samples should not be editable on the Line List Page

### PR DESCRIPTION
**Pull request by joshsadam**
screen shot 2018-11-21 at 10 25 02 am

Fixes #198

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java [hunk 1]
```diff
@@ -61,10 +68,16 @@ public LineListController(ProjectService projectService, SampleService sampleSer
...
+		Authentication authentication = SecurityContextHolder.getContext()
+				.getAuthentication();
 		Project project = projectService.read(projectId);
+
 		List<Join<Project, Sample>> projectSamples = sampleService.getSamplesForProject(project);
 		return projectSamples.stream()
-				.map(this::formatSampleMetadata)
+				.map(join -> {
+					ProjectSampleJoin psj = (ProjectSampleJoin)join;
+					return new UISampleMetadata(psj, !updateSamplePermission.isAllowed(authentication, psj.getObject()));
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java]: Definitely my mistake, sorry.  Fixed in a9492467cbe7d43df3511596a0d6c36f216aaf45

File: src/main/webapp/resources/js/pages/projects/linelist/reducers/fields.js [hunk 2]
```diff
@@ -49,14 +39,33 @@ const dateColumn = {
...
+    Object.assign(col, {
+      cellRenderer: "SampleNameRenderer"
+    });
   }
+  /*
+  Set editability of the cell
+   */
+  Object.assign(col, {
+    editable: params => {
+      if (!JSON.parse(params.data.owner)) {
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/linelist/reducers/fields.js]: Makes sense: 2f8d09388fd78f40ddab548b8c294b7a30dca327

File: src/test/java/ca/corefacility/bioinformatics/irida/ria/unit/web/LineListControllerTest.java [hunk 3]
```diff
@@ -41,11 +44,4 @@ public void testGetProjectMetadataTemplateFields() {
 		verify(projectService, times(1)).read(projectId);
 		verify(metadataTemplateService, times(1)).getMetadataFieldsForProject(any(Project.class));
 	}
-
-	@Test
-	public void testGetAllProjectMetadataEntries() {
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/test/java/ca/corefacility/bioinformatics/irida/ria/unit/web/LineListControllerTest.java]: Sorry I removed that when I switched to that other service call and meant to return it: 06f7b13abcbe3e4e94d24239e14a14c34c57214e

File: src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/projects/ProjectLineListPageIT.java [hunk 4]
```diff
@@ -50,27 +51,27 @@ public void testTableSetup() {
 
 		// Toggle one of the fields and make sure the table updates;
 		page.toggleMetadataField(1);
-		assertEquals("Should now only display 5 fields", 5, page.getNumberOfTableColumnsVisible());
+		assertEquals("Should now only display 6 fields", 6, page.getNumberOfTableColumnsVisible());
 		page.toggleMetadataField(2);
-		assertEquals("Should now only display 4 fields", 4, page.getNumberOfTableColumnsVisible());
+		assertEquals("Should now only display 5 fields", 5, page.getNumberOfTableColumnsVisible());
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/projects/ProjectLineListPageIT.java]: Test added in 7bc04924c9d2169cd9e6389dd2cf742dc6740573

File: src/main/webapp/resources/js/pages/projects/linelist/reducers/fields.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tom114) [reply src/main/webapp/resources/js/pages/projects/linelist/reducers/fields.js]: I see the change to `||` in  7bc0492.  Seems to work well that way.  Thanks.

**Pull request merged**